infercnvpy/tl/_copykat.py

```diff
@@ -0,0 +1,106 @@
+    Bayesian approaches to identify genome-wide aneuploidy at 5MB resolution 
+    in single cells to separate tumor cells from normal cells, and tumor 
+    subclones using high-throughput sc-RNAseq data.
+    
+    Note: The matrix values are often the count of unique molecular identifier (UMI) 
+    from nowadays high througput single cell RNAseq data. The early generation of 
+    scRNAseq data may be summarized as TPM values or total read counts, 
+    which should also work.
+
+    You can find more info on GitHub: https://github.com/navinlabcode/copykat
+
+    DOI: https://doi.org/10.1038/s41587-020-00795-2
+
+    Parameters
+    ----------
+    adata
+        annotated data matrix
+    key_added
+        Key under which the copyKAT scores will be stored in `adata.obs`.
+    inplace
+        If True, store the result in adata, otherwise return it.
+
+    Returns
+    -------
+    Depending on the value of `inplace`, either returns `None` or a vector
+    with scores.
+    """
+    try:
+        from rpy2.robjects.packages import importr
+        from rpy2.robjects import pandas2ri, numpy2ri
+        from rpy2.robjects.conversion import localconverter
+        from rpy2 import robjects as ro
+    except ImportError:
+        raise ImportError("copyKAT requires rpy2 to be installed. ")
+
+    try:
+        copyKAT = importr("copykat")
+    except ImportError:
+        raise ImportError(
+            "copyKAT requires a valid R installation with the following packages: "
+            "copykat"
+        )
+    
+    logging.info("Preparing R objects")
+    with localconverter(ro.default_converter + numpy2ri.converter):
+        expr = adata.X if layer is None else tmp_adata.layers[layer]
+        if issparse(expr):
+            expr = expr.T.toarray()
+        else:
+            expr = expr.T
+        ro.globalenv["expr_r"] = ro.conversion.py2rpy(expr)
+    ro.globalenv["gene_names"] = ro.conversion.py2rpy(list(adata.var.index))
+    ro.globalenv["cell_IDs"] = ro.conversion.py2rpy(list(adata.obs.index))
+    
+    logging.info("Running copyKAT")
+    ro.r(
+        """
+        rownames(expr_r) <- gene_names
+        colnames(expr_r) <- cell_IDs
+        copyKAT_run <- copykat(rawmat = expr_r, id.type = "S", ngene.chr = 5, win.size = 25, 
+                                KS.cut = 0.1, sam.name = "test", distance = "euclidean", norm.cell.names = "", 
+                                n.cores = 12, output.seg = FALSE)
+        copyKAT_result <- copyKAT_run$CNAmat
+        colnames(copyKAT_result) <- c('chrom', 'chrompos', 'abspos', cell_IDs)
+        """
+    )
+
+    with localconverter(
+        ro.default_converter + numpy2ri.converter + pandas2ri.converter
+    ):
+        copyKAT_result = ro.conversion.rpy2py(ro.globalenv["copyKAT_result"])
+
+    adata.uns["copyKAT"] = {
+        "chr_pos": {
+            f"chr{chrom}": int(pos)
+            for pos, chrom in copyKAT_result.loc[:, ["chrom"]].drop_duplicates().itertuples()
+        }
+    }
```

Comment:
Please store this in a temporary variable and only assign it to `adata` if `inplace == True`

Reply:
and done